nep-12.mediawiki

```diff
@@ -0,0 +1,2098 @@
+* Sign transaction upon receiving confirmation from user
+* If the "broadcastOverride" input argument is set to True, return the serialized signed transaction to the dApp
+* Else broadcast the RPC request
+* Return the transaction Id and the url of the rpc node which the tx was broadcast to
+
+Example
+<pre>
+{
+  txid: 'ed54fb38dff371be6e3f96e4880405758c07fe6dd1295eb136fe15f311e9ff77',
+  nodeUrl: 'http://seed7.ngd.network:10332',
+}
+</pre>
+
+
+===Events===
+
+dApps can listen for events emitted by the wallet provider using the `addEventListener` method.
+
+====addEventListener====
+
+'''Method Interface'''
+
+<pre>
+function addEventListener(event: Event, callback: Function): Void
+</pre>
+
+'''Input arguments'''
+
+<pre>
+enum Event {
+  READY = 'READY',
+  ACCOUNT_CHANGED = 'ACCOUNT_CHANGED',
+  NETWORK_CHANGED = 'NETWORK_CHANGED',
+  CONNECTED = 'CONNECTED',
+  DISCONNECTED = 'DISCONNECTED',
+  BLOCK_HEIGHT_CHANGED = 'BLOCK_HEIGHT_CHANGED',
+  TRANSACTION_CONFIRMED = 'TRANSACTION_CONFIRMED'
+}
+</pre>
+
+=====READY=====
+
+On a READY event, the callback will fire with a single argument with information about the wallet provider. At any time a READY event listener is added, it will immidiately be called if the provider is already in a ready state. This provides a single flow for dapp developers since this listener should start any and all interactions with the dapi protocol.
+<pre>
+interface Provider {
+  name: string;
+  website: string;
+  version: string;
+  compatibility: string[];
+}
+</pre>
+
+=====ACCOUNT_CHANGED=====
+
+On a ACCOUNT_CHANGED event, the callback will fire with a single argument of the new account. This occurs when an account is already connected to the dapp, and the user has changed the connected account from the dapi provider side.
+<pre>
+interface Account {
+  address: string; // Address of the connected account
+  label?: string; // A label the users has set to identify their wallet
+}
+</pre>
+
+=====CONNECTED=====
+On a CONNECTED event, the user has approved the connection of the dapp with one of their accounts. This will fire the first time any of one of the following methods are called from the dapp: getAccount, invoke, send.
+
+<pre>
+interface Account {
+  address: string; // Address of the connected account
+  label?: string; // A label the users has set to identify their wallet
+}
+</pre>
+
+
+=====DISCONNECTED=====
+
+On a DISCONNECTED event, the account connected to the dapp via the dapi provider has been disconnected (logged out).
+
+
+=====NETWORK_CHANGED=====
+
+On a NETWORK_CHANGED event, the callback will fire with a single argument of the new network details. This occurs when the user has changed the network on their provider wallet.
+
+<pre>
+interface Networks {
+  networks: string[]; // Array of network names the wallet provider has available for the dapp developer to connect to.
+  defaultNetwork: string; // Network the wallet is currently set to.
```

Comment:
Now I got the idea, the default is the same as the current.

Should we change the name: from `defaultNetwork` to `currentNetwork`.

And the end, it will be the same...it is up to you.

Reply:
Agree it could be `currentNetwork`, but `defaultNetwork` seems to be working for now.